config/routes.rb

```diff
@@ -1,21 +1,21 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  devise_for :users, controllers: {
+  devise_for :users, skip: :registrations, controllers: {
```

Comment:
I skipped registrations to get rid of the default devise registration links like /users/sign_in as we have implemented our own registration workflow